qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -45,6 +51,85 @@ class UndoEntry:
     pinned = attr.ib()
 
 
+class TabDeque:
+    """Class which manages the 'last visited' tab stack.
+
+    Instead of handling deletions by clearing old entries, they are handled by
+    checking if they exist on access. This allows us to save an iteration on
+    every tab delete.
+
+    Currently, we assume we will switch to the tab returned by any of the
+    getter functions. This is done because the tab_switch functions will be
```

Comment:
Should this be `on_switch` instead?

Also: you can check if the going to be added tab is the same as the stack present and not add it in that case.